### PR TITLE
fix websocket hang

### PR DESCRIPTION
**Pull request by dylan-conway**

### CODE REVIEW

File: src/bun.js/bindings/webcore/WebSocket.cpp [hunk 1]
```diff
@@ -193,6 +194,7 @@ WebSocket::~WebSocket()
         break;
     }
     }
+    deref();
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/webcore/WebSocket.cpp]: Doesn't deref potentially call `delete` and destruct the WebSocket? is it defined behavior in C++ to call a destructor inside a destructor?

Comment (dylan-conway) [reply src/bun.js/bindings/webcore/WebSocket.cpp]: You're right, pretty sure it's unnecessary. Tests still do not hang without it

File: test/bun.js/websocket.test.js [hunk 2]
```diff
@@ -131,3 +133,50 @@ describe("WebSocket", () => {
...
+        }
+
+        return new Response("http response");
+      },
+      websocket: {
+        open(ws) {
+          ws.send("hello websocket");
+        },
+        message(ws, message) {
+          expect(message).toEqual("hello");
```
Comment (Jarred-Sumner) [inline test/bun.js/websocket.test.js]: can you set a boolean that this function was called in the test

Comment (dylan-conway) [reply test/bun.js/websocket.test.js]: updated

**Comment by Jarred-Sumner** [comment]
can you add more tests that check we ref/unref correctly in the `"error"` handler, when there are no handlers, etc?

**Pull request merged**